src/lib/storage/base_segment.hpp

```diff
@@ -37,6 +38,22 @@ class BaseSegment : private Noncopyable {
   // such as strings who memory usage is implementation defined
   virtual size_t estimate_memory_usage() const = 0;
 
+  const std::optional<OrderByMode> sort_order() const;
```

Comment:
Return values shouldn't impose constness on the caller, please remove the `const` from the return type